os/arch/arm/src/stm32l4/stm32l4_allocateheap.c

```diff
@@ -224,46 +260,68 @@ void up_allocate_heap(FAR void **heap_start, size_t *heap_size)
    */
 
   uintptr_t ubase = (uintptr_t)USERSPACE->us_bssend + CONFIG_MM_KERNEL_HEAPSIZE;
-  size_t    usize = SRAM1_END - ubase;
-  int       log2;
 
-  DEBUGASSERT(ubase < (uintptr_t)SRAM1_END);
+  /* Is ubase in SRAM1 region? */
 
-  /* Adjust that size to account for MPU alignment requirements.
-   * NOTE that there is an implicit assumption that the SRAM1_END
-   * is aligned to the MPU requirement.
-   */
+  if (ubase < SRAM1_END) {
+    /* Yes, let's register remain region as a heap. */
 
-  log2  = (int)mpu_log2regionfloor(usize);
-  DEBUGASSERT((SRAM1_END & ((1 << log2) - 1)) == 0);
+    size_t    usize = SRAM1_END - ubase;
+    int       log2;
 
-  usize = (1 << log2);
-  ubase = SRAM1_END - usize;
+    /* Adjust that size to account for MPU alignment requirements.
+     * NOTE that there is an implicit assumption that the SRAM1_END
+     * is aligned to the MPU requirement.
+     */
 
-  /* Return the user-space heap settings */
+    log2  = (int)mpu_log2regionfloor(usize);
+    DEBUGASSERT((SRAM1_END & ((1 << log2) - 1)) == 0);
 
-  board_led_on(LED_HEAPALLOCATE);
-  *heap_start = (FAR void *)ubase;
-  *heap_size  = usize;
+    usize = (1 << log2);
+    ubase = SRAM1_END - usize;
 
-  /* Colorize the heap for debug */
+    /* Return the user-space heap settings */
+
+    board_led_on(LED_HEAPALLOCATE);
+    *heap_start = (FAR void *)ubase;
+    *heap_size  = usize;
 
-  up_heap_color((FAR void *)ubase, usize);
+    /* Colorize the heap for debug */
 
-  /* Allow user-mode access to the user heap memory */
+    up_heap_color((FAR void *)ubase, usize);
```

Comment:
why don't you use 'heap_start' instead of  '(FAR void  *)usbase'?

Reply:
In protected build case, it starts from base address of user space. That's why it named as ubase.

Reply:
And it's a chip-specific file, we had not better modify it by ourself if it is not wrong.